app/models/post.rb

```diff
@@ -0,0 +1,20 @@
+class Post < ApplicationRecord
+  has_many :favorites, dependent: :destroy
+  has_many :users, through: :favorites
+
+  def update_hacker_news_post
+    hn = HackerNews.new
+    self.hacker_news_post_id = hn.story(hacker_news_post_id)['id']
+    self.title = hn.story(hacker_news_post_id)['title']
+    self.score = hn.story(hacker_news_post_id)['score']
+    self.by = hn.story(hacker_news_post_id)['by']
+    self.time = Time.at(hn.story(hacker_news_post_id)['time'])
+    self.url = hacker_news_url(hn.story(hacker_news_post_id)['url'])
+  end
+
+  def hacker_news_url(url)
+    return url unless url.nil?
+
+    "https://news.ycombinator.com/item?id=#{hacker_news_post_id}"
+  end
```

Comment:
Protect against the edge case of a URL being nil. HackerNews sometimes sends a nil URL when it points to an item on their website (ex: list of jobs)